filebeat/channel/connector.go

```diff
@@ -51,22 +54,28 @@ func (c *pipelineConnector) ConnectWith(cfg *common.Config, clientCfg beat.Clien
 		return nil, err
 	}
 
-	var err error
-	var userProcessors beat.ProcessorList
+	procs := processors.NewList(nil)
 
-	userProcessors, err = processors.New(config.Processors)
+	userProcessors, err := processors.New(config.Processors)
 	if err != nil {
 		return nil, err
 	}
+	procs.List = append(procs.List, userProcessors)
+
+	if !config.Index.IsEmpty() {
+		staticFields := fmtstr.FieldsForBeat(
+			c.parent.beatInfo.Beat, c.parent.beatInfo.Version)
+		timestampFormat, err :=
+			fmtstr.NewTimestampFormatString(&config.Index, staticFields)
+		if err != nil {
+			return nil, err
+		}
+		indexProcessor := &addFormattedIndex{timestampFormat}
+		procs.List = append(procs.List, indexProcessor)
+	}
 
 	if lst := clientCfg.Processing.Processor; lst != nil {
-		if len(userProcessors.All()) == 0 {
-			userProcessors = lst
-		} else if orig := lst.All(); len(orig) > 0 {
-			newLst := processors.NewList(nil)
-			newLst.List = append(newLst.List, lst, userProcessors)
-			userProcessors = newLst
-		}
+		procs.List = append(procs.List, lst)
```

Comment:
This changes order. The userProcessors are supposed to be appended after 'lst'. Unfortunately processors API has been somewhat inconsistent in the past, which did force us to be 'clever' here :(